Add custom style property for progress value

tonygoldcrest opened the pull request:
Connected to #11.

1) Foreground is inside background now.
2) Removed reflectToAttribute.
3) Added custom style property for progress value.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-progress-bar/14)
<!-- Reviewable:end -->

Commit:
Changes in progress-bar.

1) Foreground is inside background now.
2) Removed reflectToAttribute.
3) Added custom style property for progress value.

manolo commented:
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[vaadin-progress-bar.html, line 40 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-KvYQ-aUI3FZVfhGztUJ:-KvYQ-aUI3FZVfhGztUK:bdzdylp) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/5ff3565a57c686cd52bc5e635e6d630520688754/vaadin-progress-bar.html#L40)):*
> ```HTML
>         transition: transform 150ms;
>         width: 100%;
>         transform: scaleX(var(--vaadin-progress-value));
> ```

What this custom property means? you need to document in element jsdoc.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

limonte commented:
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---

*[vaadin-progress-mixin.html, line 44 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-Kvaw1BzGY4H3rg3gquw:-Kvaw1BzGY4H3rg3gqux:b-2ieww1) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/5ff3565a57c686cd52bc5e635e6d630520688754/vaadin-progress-mixin.html#L44)):*
> ```HTML
>       _valueChanged(newV, oldV) {
>         this.updateStyles({
>           '--vaadin-progress-value': this._normalizeValue(newV, this.min, this.max)
> ```

Unit test needed. Tip: you can use https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/cssText

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

Commit:
Added and fixed tests.
Renamed background and foreground to bar and value.
Added documentation for progress-value variable.

tonygoldcrest commented:
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-progress-bar.html, line 40 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-KvYQ-aUI3FZVfhGztUJ:-KvbSjlFRx5vwxpUV0NR:b-896fix) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/5ff3565a57c686cd52bc5e635e6d630520688754/vaadin-progress-bar.html#L40)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

What this custom property means? you need to document in element jsdoc.
</blockquote></details>

Done.

---

*[vaadin-progress-mixin.html, line 44 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-Kvaw1BzGY4H3rg3gquw:-KvbSl-HoMkm69PhqHws:b-896fix) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/5ff3565a57c686cd52bc5e635e6d630520688754/vaadin-progress-mixin.html#L44)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

Unit test needed. Tip: you can use https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/cssText
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[test/vaadin-progress-bar.html, line 26 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-KvbSyFvq4zz0_LwJ1JF:-KvbSyFwN7sPTKUUYarp:b-e4e0cg) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/d05a84b2377823f50eba3d81b809920e6a126e74/test/vaadin-progress-bar.html#L26)):*
> ```HTML
>         progress.value = 0.1;
> 
>         setTimeout(() => {
> ```

Ah, you'll need to add "done" parameter for the function so it's a valid asynchronous test. Otherwise the async block might not get invoked at all.
```
it('should have proper scale', done => {
...
requestAnimationFrame(() => {
  expect...
  done();
});
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

Commit:
Removed animation and async tests.

tonygoldcrest commented:
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[test/vaadin-progress-bar.html, line 26 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-KvbSyFvq4zz0_LwJ1JF:-KvbaMlxveCg6WiyGXtd:b-896fix) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/d05a84b2377823f50eba3d81b809920e6a126e74/test/vaadin-progress-bar.html#L26)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Ah, you'll need to add "done" parameter for the function so it's a valid asynchronous test. Otherwise the async block might not get invoked at all.
```
it('should have proper scale', done => {
...
requestAnimationFrame(() => {
  expect...
  done();
});
```
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 2 files at r1, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-:-KvbbTFbFU1zyQZ2wzt2:bnfp4nl)*
<!-- Sent from Reviewable.io -->

limonte commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 2 files at r1, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-:-KvbdoqeXYDyyjsTs64D:bnfp4nl)*
<!-- Sent from Reviewable.io -->

Commit:
Remove styling test.

jouni commented on vaadin-progress-bar.html:
```diff
@@ -23,40 +24,28 @@
-    <div id="foreground" style$="transform: scaleX([[_normalizeValue(value, min, max)]])"></div>
+    <div id="bar">
+      <div id="value"></div>
+    </div>
```
Should use `part` instead of `id`, right? Or is that coming in another PR? In any case, remember to document the stylable parts.

tomivirkki replied:
That will be done in a separate PR, right?

tonygoldcrest commented:
Review status: 2 of 5 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[vaadin-progress-bar.html, line 48 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-KvXUmS3hUEcq_83irA0-r4-48:-KvfgocRhxdXVq7HOX_-:by7z77) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/7541f20f7dabfbdd762858b431f2084f9b68a7aa/vaadin-progress-bar.html#L48)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

That will be done in a separate PR, right?
</blockquote></details>

Yes, with themable mixin.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

manolo commented:
Visual tests seem failing, could you verify ? 

---

Reviewed 1 of 2 files at r1, 1 of 2 files at r3, 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-:-Kvfi1sEp5eWLB7LzwJz:bhtamqk)*
<!-- Sent from Reviewable.io -->

Commit:
Update visual test for edge.

tomivirkki committed:
Fix a test

manolo commented:
Still failing in CI

---

Review status: 4 of 6 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-:-KvgrFDWqGsIC5ldo1vO:b-14nogo)*
<!-- Sent from Reviewable.io -->

manolo commented:
Reviewed 1 of 1 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[test/visual/screens/vaadin-progress-bar/default-tests/vaadin-progress-bar/edge.png, line 0 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-KvgsgivG7shBxuBV5TK:-KvgsgivG7shBxuBV5TL:bf09hea) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/86f6ce0f49a1a5754d94b464006a594763d9dfed/test/visual/screens/vaadin-progress-bar/default-tests/vaadin-progress-bar/edge.png#L0)):*
This was actually Gemini spotting an issue. The value isn't applied on Edge. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-progress-mixin.html, line 21 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-Kvgtp4SLkjQQ1FsA5vs:-Kvgtp4SLkjQQ1FsA5vt:b3srudc) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/002575783a71a08e8752d21e00e6a2456575957b/vaadin-progress-mixin.html#L21)):*
> ```HTML
>           value: {
>             type: Number,
>             observer: '_valueChanged'
> ```

We'll probably need to use multi-property observers for value, min and max since the order in which they're applied can be anything and min/max affect value anyway.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

Commit:
Added multi-value observer for value, min and max properties.

tonygoldcrest commented:
Review status: 5 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-progress-mixin.html, line 21 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-Kvgtp4SLkjQQ1FsA5vs:-Kw-C_xUdA6m60XT3C7A:bs7wumb) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/002575783a71a08e8752d21e00e6a2456575957b/vaadin-progress-mixin.html#L21)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

We'll probably need to use multi-property observers for value, min and max since the order in which they're applied can be anything and min/max affect value anyway.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

Commit:
Fixed issue in Edge.

tonygoldcrest commented:
Review status: 5 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[test/visual/screens/vaadin-progress-bar/default-tests/vaadin-progress-bar/edge.png, line  at r5](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-KvgsgivG7shBxuBV5TK:-Kw-KycaVmLs1OiRkYMR:b-896fix) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/86f6ce0f49a1a5754d94b464006a594763d9dfed/test/visual/screens/vaadin-progress-bar/default-tests/vaadin-progress-bar/edge.png#L)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

This was actually Gemini spotting an issue. The value isn't applied on Edge. 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[test/visual/screens/vaadin-progress-bar/default-tests/vaadin-progress-bar/edge.png, line  at r5](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-KvgsgivG7shBxuBV5TK:-Kw-LEIWGzbY14ssMquh:b-uqkwa2) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/86f6ce0f49a1a5754d94b464006a594763d9dfed/test/visual/screens/vaadin-progress-bar/default-tests/vaadin-progress-bar/edge.png#L)):*
<details><summary><i>Previously, Toshegg wrote…</i></summary><blockquote>

Done.
</blockquote></details>

The original reference image should be restored now that the value works on Edge.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

Commit:
Update visuals.

tonygoldcrest commented:
Review status: 4 of 5 files reviewed at latest revision, 3 unresolved discussions.

---

*[test/visual/screens/vaadin-progress-bar/default-tests/vaadin-progress-bar/edge.png, line  at r5](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-KvgsgivG7shBxuBV5TK:-Kw-MGmGhKokN-N6s5Vn:b-896fix) ([raw file](https://github.com/vaadin/vaadin-progress-bar/blob/86f6ce0f49a1a5754d94b464006a594763d9dfed/test/visual/screens/vaadin-progress-bar/default-tests/vaadin-progress-bar/edge.png#L)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

The original reference image should be restored now that the value works on Edge.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 3 files at r4, 1 of 1 files at r6, 1 of 2 files at r7.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-:-Kw-NPMe4cYQFRTtKnih:bnfp4nl)*
<!-- Sent from Reviewable.io -->

manolo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 2 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-progress-bar/14#-:-Kw-n9hHQiXXNa05AfkO:bnfp4nl)*
<!-- Sent from Reviewable.io -->
